geometry/cartesian_product.hpp

```diff
@@ -0,0 +1,34 @@
+
+#pragma once
+
+#include <algorithm>
+#include <tuple>
+#include <utility>
+
+namespace principia {
+namespace geometry {
+namespace internal_cartesian_product {
+
+// This struct exports two functions:
+//
+//   static constexpr ... Add(LTuple const& left, RTuple const& right);
+//   static constexpr... Subtract(LTuple const& left, RTuple const& right);
+//
+// where the return types are appropriate for the operation.  The two tuples may
+// be of different types (including different sizes) but corresponding element
+// types must have an addition and a subtraction.
```

Comment:
Document the semantics of adding tuples of different sizes (the smaller one is zero-extended, where zero is the neutral element of addition on the relevant types).

Reply:
Done.